src/rgw/rgw_sal_motr.cc

```diff
@@ -2192,29 +2195,63 @@ int MotrObject::read_mobj(const DoutPrefixProvider* dpp, int64_t off, int64_t en
 
 int MotrObject::get_bucket_dir_ent(const DoutPrefixProvider *dpp, rgw_bucket_dir_entry& ent)
 {
-  int rc = 0;
+  int rc = 0, ret_code = -ENOENT;
   string tenant_bkt_name = get_bucket_name(this->get_bucket()->get_tenant(), this->get_bucket()->get_name());
   string bucket_index_iname = "motr.rgw.bucket.index." + tenant_bkt_name;
-  int max = 1000;
+  int max = 2;
   vector<string> keys(max);
   vector<bufferlist> vals(max);
   bufferlist bl;
   bufferlist::const_iterator iter;
 
-  if (this->get_bucket()->get_info().versioning_status() == BUCKET_VERSIONED ||
-      this->get_bucket()->get_info().versioning_status() == BUCKET_SUSPENDED) {
+  // Check entry in the cache
+  if (this->store->get_obj_meta_cache()->get(dpp, this->get_name(), bl) == 0) {
+      iter = bl.cbegin();
+      ent.decode(iter);
+      rc = 0;
+      goto out;
+  }
+  if(this->have_instance())
+  {
+    if(this->get_key().have_null_instance())
+      {
+        // Read the null index entry
+        std::string null_version_key = this->get_name() + "/null";
+        bufferlist bl_null;
+        bufferlist::const_iterator iter1;
+        rgw_bucket_dir_entry null_key;
+        ret_code = this->store->do_idx_op_by_name(bucket_index_iname,
```

Comment:
First do the lookup for null in the cache,  if not found then lookup in the index.

Reply:
done